src/migrations/20171129064057_multicontainer.js

```diff
@@ -0,0 +1,311 @@
+
+var defaultLegacyVolume = function (appId) {
+	return `resin-app-${appId}`
+}
+
+var tryParse = function (obj) {
+	try {
+		return JSON.parse(obj)
+	}
+	catch(e) {
+		return {}
+	}
+}
+
+var singleToMulticontainerApp = function (app) {
+	// From *very* old supervisors, env or config may be null
+	// so we ignore errors parsing them
+	let conf = tryParse(app.config)
+	let env = tryParse(app.env)
+	let environment = {}
+	let appId = parseInt(app.appId)
+	for (let key in env) {
+		if (!/^RESIN_/.test(key)) {
+			environment[key] = env[key]
+		}
+	}
+	let newApp = {
+		appId: appId,
+		commit: app.commit,
+		name: app.name,
+		releaseId: 1,
+		networks: {},
+		volumes: {}
+	}
+	let defaultVolume = defaultLegacyVolume(appId)
+	newApp.volumes[defaultVolume] = {}
+	let updateStrategy = conf['RESIN_SUPERVISOR_UPDATE_STRATEGY']
+	if (updateStrategy == null) {
+		updateStrategy = 'download-then-kill'
+	}
+	let handoverTimeout = conf['RESIN_SUPERVISOR_HANDOVER_TIMEOUT']
+	if (handoverTimeout == null) {
+		handoverTimeout = ''
+	}
+	let restartPolicy = conf['RESIN_APP_RESTART_POLICY']
+	if (restartPolicy == null) {
+		restartPolicy = 'always'
+	}
+	newApp.services = [
+		{
+			serviceId: 1,
+			appId: appId,
+			serviceName: app.name.toLowerCase(),
+			imageId: 1,
+			commit: app.commit,
+			releaseId: 1,
+			image: app.imageId,
+			privileged: true,
+			networkMode: 'host',
+			volumes: [
+				`${defaultVolume}:/data`
+			],
+			labels: {
+				'io.resin.features.kernel-modules': '1',
+				'io.resin.features.firmware': '1',
+				'io.resin.features.dbus': '1',
+				'io.resin.features.supervisor-api': '1',
+				'io.resin.features.resin-api': '1',
+				'io.resin.update.strategy': updateStrategy,
+				'io.resin.update.handover-timeout': handoverTimeout,
+				'io.resin.legacy-container': '1'
```

Comment:
The cli and builder do not set this label when transforming legacy apps to multicontainer, should they?

Reply:
No need, this is just for the supervisor to know it has to ignore the serviceId difference when comparing if nothing else changed (since we didn't have serviceId for old containers)

Reply:
And actually I realized it's also not relevant  here. Only on lib/migration.coffee, for legacy preloaded apps.